src/map/script.cpp

```diff
@@ -25686,6 +25722,281 @@ BUILDIN_FUNC( openstylist ){
+
+		if (id == nullptr) {
+			ShowError("buildin_setitemdata: SID_ID: Nonexistant item '%s'.\n", script_getstr(st, 4));
+			st->state = END;
+			return SCRIPT_CMD_FAILURE;
+		}
+		if (!itemdb_isequip2( id.get() ) || id->type == IT_AMMO) {
+			ShowError("buildin_setitemdata: SID_ID: Invalid type for item '%s'. The item must be equippable.\n", script_getstr(st, 4));
+			st->state = END;
+			return SCRIPT_CMD_FAILURE;
+		}
+		itm->nameid = id->nameid;
+		break;
+	}
+	case SID_IDENTIFY:
+		itm->identify = script_getnum(st, 4) > 0;
+		break;
+	case SID_REFINE: {
+		int refine = script_getnum(st, 4);
+		if (refine < 0 || refine > MAX_REFINE) {
+			ShowError("buildin_setitemdata: SID_REFINE: Refine %d is out of the range 0-%d.\n", refine, MAX_REFINE);
+			st->state = END;
+			return SCRIPT_CMD_FAILURE;
+		}
+		itm->refine = refine;
+		break;
+	}
+	case SID_ATTRIBUTE:
+		itm->attribute = script_getnum(st, 4) > 0;
+		break;
+	case SID_CARD1:
+	case SID_CARD2:
+	case SID_CARD3:
+	case SID_CARD4: {
+		int index = type - SID_CARD1;
+
+		t_itemid card;
+
+		if (script_isstring(st, 4)) {
+			std::shared_ptr<item_data> id = item_db.searchname( script_getstr(st, 4) );
+
+			if (id == nullptr) {
+				ShowError("buildin_setitemdata: SID_CARD%d: Nonexistant item '%s'.\n", index + 1, script_getstr(st, 4));
+				st->state = END;
+				return SCRIPT_CMD_FAILURE;
+			}
+			card = id->nameid;
+		}
+		else {
+			card = script_getnum(st, 4);
+			if (card != 0 && !item_db.exists(card) && ((index == 0 && !itemdb_isspecial(card)) || (index > 0 && !itemdb_isspecial(itm->card[0])))) {
+				ShowError("buildin_setitemdata: SID_CARD%d: Nonexistant item '%u'.\n", index + 1, card);
+				st->state = END;
+				return SCRIPT_CMD_FAILURE;
+			}
+		}
+
+		itm->card[index] = card;
+		break;
+	}
+	case SID_EXPIRE: {
+		int t = script_getnum(st, 4);
+		if (t <= 0) {
+			ShowError("buildin_setitemdata: SID_EXPIRE: Invalid expire time %d. The time can't be below 1.\n", t);
+			st->state = END;
+			return SCRIPT_CMD_FAILURE;
+		}
+		itm->expire_time = (unsigned int)(time(NULL) + t);
+		break;
+	}
+	case SID_BOUND: {
+		int bound = script_getnum(st, 4);
+		if (bound < BOUND_NONE || bound >= BOUND_MAX) {
+			ShowError("buildin_setitemdata: SID_BOUND: %d is not a correct bound type.\n", bound);
+			st->state = END;
+			return SCRIPT_CMD_FAILURE;
+		}
+		itm->bound = static_cast<bound_type>( bound );
+		break;
+	}
+	case SID_GRADE: {
+		int grade = script_getnum(st, 4);
+		if (grade < 0 || grade > MAX_ENCHANTGRADE) {
+			ShowError("buildin_setitemdata: SID_GRADE: Grade %d is out of the range 0-%d.\n", grade, MAX_ENCHANTGRADE);
+			st->state = END;
+			return SCRIPT_CMD_FAILURE;
+		}
+		itm->enchantgrade = grade;
+		break;
+	}
+	case SID_OPTIONID1:
+	case SID_OPTIONID2:
+	case SID_OPTIONID3:
+	case SID_OPTIONID4:
+	case SID_OPTIONID5: {
+		int index = type - SID_OPTIONID1;
+
+		t_itemid id = script_getnum(st, 4);
+
+		if (!random_option_db.exists(id)) {
```

Comment:
Maybe here we should also add a "special case" for id = 0 (which would also need to remove optionvalue and optionparam of the same index)?